esphome/components/ccs811/ccs811.h

```diff
@@ -43,6 +44,7 @@ class CCS811Component : public PollingComponent, public i2c::I2CDevice {
 
   sensor::Sensor *co2_{nullptr};
   sensor::Sensor *tvoc_{nullptr};
+  sensor::Sensor *version_{nullptr};
```

Comment:
```suggestion
  text_sensor::TextSensor *version_{nullptr};
```

Nit: A version number should really be published as a text sensor - will require a couple of smaller changes too:

* Add `AUTO_LOAD = ["text_sensor"]` to `ccs811/sensor.py`
* Convert `sensor.sensor_schema` to use `text_sensor.TEXT_SENSOR_SCHEMA`
* Use `sens = cg.new_Pvariable(config[CONF_VERSION][CONF_ID])` and `await text_sensor.register_text_sensor(sens, config[CONF_VERSION])` to declare the text sensor.
* Publish state using `publish_state` as before, but instead pass in a formatted `std::string`

Reply:
I was overthinking your list of changes but @ssieb put me back on the right path. It's a text sensor now.